modes/mu4e/evil-collection-mu4e.el

```diff
@@ -93,6 +93,31 @@
       (evil-set-initial-state mode 'normal))
     (evil-set-initial-state 'mu4e-compose-mode 'insert))
 
+  ;; These functions were removed from mu4e 1.12.0, specifically this commit:
+  ;; https://github.com/djcb/mu/commit/85bfe763362b95935a3967f6585e14b3f9890a70
+  (when (version<= "1.12.0" mu4e-mu-version)
+    (defun mu4e-compose-goto-top ()
+      "Go to the beginning of the message or buffer.
+Go to the beginning of the message or, if already there, go to the
+beginning of the buffer."
+      (interactive)
+      (let ((old-position (point)))
+        (message-goto-body)
+        (when (= (point) old-position)
+          (goto-char (point-min)))))
+
+    (defun mu4e-compose-goto-bottom ()
+      "Go to the end of the message or buffer.
+Go to the end of the message (before signature) or, if already there, go to the
+end of the buffer."
+      (interactive)
+      (let ((old-position (point))
+            (message-position (save-excursion (message-goto-body) (point))))
+        (end-of-buffer)
+        (when (re-search-backward "^-- $" message-position t)
```

Comment:
Why not use `message-goto-signature`, like how you used `message-goto-body` above? It also takes into account that `message-signature-separator` can be different from `--`.

Reply:
Yes, I copied the code for those Mu4e commands verbatim from the original source. Now I have replaced that `^-- $` with `message-signature-separator`. It's similar to `message-goto-signature` except it goes to the line above the signature rather than the line below and it also does not push the mark. This seems like better behavior to me for Evil but I am interested in what you think about it now.

Reply:
Haven't tested it but seems fine to me :+1: 

Btw, I was thinking, then doesn't `message-goto-body` used in `mu4e-compose-goto-top` also push the mark? Checked the source code and it has a special clause that it only pushes the mark if called interactively, but `message-goto-signature` doesn't have that, which tbh. seems inconsistent behavior to me...